tests/functions/test_encode_categorical.py

```diff
@@ -74,10 +74,10 @@ def test_encode_categorical_invalid_input(df):
 @settings(deadline=None)
 def test_encode_categorical_invalid_input_2(df):
     """
-    Raise TypeError for wrong input type
+    Raise KeyError for wrong input type
     for `column_names`.
     """
-    with pytest.raises(TypeError):
+    with pytest.raises(KeyError):
         df.encode_categorical({"names"})
 
```

Comment:
any way we could support this instead of letting it throw an error? seems like we should be able to accept any generic iterable (like sets), at least in this case.